src/rebar_prv_eunit.erl

```diff
@@ -121,9 +121,9 @@ add_test_dir(Opts) ->
     %% if no src_dirs are set we have to specify `src` or it won't
     %% be built
     case proplists:append_values(src_dirs, Opts) of
-        [] -> [{src_dirs, ["src", "test"]}];
-        Srcs -> [{src_dirs, ["test"|Srcs]}]
-    end ++ lists:keydelete(src_dirs, 1, Opts).
+        [] -> [{src_dirs, ["src", "test"]} | Opts];
+        _ -> [{src_dirs, ["test"]} | Opts]
+    end.
```

Comment:
Didn't we need to *merge* `"test"` with the existing `src_dirs` value if it existed or something?